File: Source/Assets/Scripts/GameLogic/Collision/EnemyCollision.js
var EnemyDamage = 10;

function Awake()
{
	GetComponent.<Renderer>().material.color = Color.red;
}

//function Collide()
function OnTriggerEnter(other:Collider)
{
	if (other.gameObject.tag == "Player")
	{
		Destroy (gameObject);
		var world = GameObject.Find("Level");
		var scoreManager = world.GetComponent(HealthManager);
		scoreManager.AddOffset(-EnemyDamage);
	}
}

File: Source/Assets/Scripts/GameLogic/Collision/EnemyCollision.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class EnemyCollision : MonoBehaviour
{
    public int EnemyDamage = 10;

    void Start()
    {
        GetComponent<Renderer>().material.color = Color.red;
    }

    //Detect collisions between the GameObjects with Colliders attached
    void OnCollisionEnter(Collision collision)
    {
        if (collision.gameObject.tag != "Player")
            return;
            
        Debug.Log("Enemy collision logic happened on " + GetInstanceID());
        Destroy (gameObject);
		GameObject world = GameObject.Find("Level");
		HealthManager scoreManager = world.GetComponent<HealthManager>();
		scoreManager.AddOffset(-EnemyDamage);        
    }
}